common/modal.js

```diff
@@ -66,11 +66,13 @@
         vm.params = params;
         vm.ok = ok;
         vm.cancel = cancel;
+        vm.submit = submitMutliSelection;
 
         var reference = params.reference;
         vm.hasLoaded = false;
         var reference = vm.reference = params.reference;
-
+        var selectMode = params.selectMode?params.selectMode:"single-select";
```

Comment:
This has been defaulted to *single-select* to support backward compatibility. (Any existing piece of code which does not leverage this attribute). It is, in my opinion, a fair assumption to have single-select in case no value is passed. 
I will move this string value to a `constant` module.